external/CMakeLists.txt

```diff
@@ -0,0 +1,72 @@
+#----------------------------------*-CMake-*----------------------------------#
+# Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+# See the top-level COPYRIGHT file for details.
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+#-----------------------------------------------------------------------------#
+
+set(_celer_external_packages)
+
+#-----------------------------------------------------------------------------#
+# GTest
+#-----------------------------------------------------------------------------#
+
+if(CELERITAS_EXTERNAL_GTest)
+  FetchContent_Declare(
+    GTest
+    URL https://github.com/google/googletest/archive/refs/tags/v1.14.0.tar.gz
+    DOWNLOAD_EXTRACT_TIMESTAMP TRUE
+  )
+  list(APPEND _celer_external_packages GTest)
+
+  set(BUILD_GMOCK OFF) # For compatibility
+  set(INSTALL_GTEST OFF)
+  set(gtest_hide_internal_symbols ON)
+  set(gtest_force_shared_crt ON) # Don't replace /MD with -MT for windows
+endif()
+
+#-----------------------------------------------------------------------------#
+# nlohmann_json
+#-----------------------------------------------------------------------------#
+
+if(CELERITAS_EXTERNAL_nlohmann_json)
+  FetchContent_Declare(
+    nlohmann_json
+    URL https://github.com/nlohmann/json/releases/download/v3.11.3/json.tar.xz
+    DOWNLOAD_EXTRACT_TIMESTAMP TRUE
```

Comment:
should we update ```cmake_minimum_required``` since this was introduced in version 3.24 (and ```FetchContent_MakeAvailable``` below in version 3.14) ?

Reply:
I think arguments like `DOWNLOAD_EXTRACT_TIMESTAMP` and `EXCLUDE_FROM_ALL` are ignored if the cmake version doesn't support them... should we set a policy instead of using extract timestamp, and I guess I can reimplement https://github.com/celeritas-project/g4vg/pull/6 ?

Reply:
I happened to have a system with an old (3.23.2) version of CMake so I tried passing ```DOWNLOAD_EXTRACT_TIMESTAMP``` and I get a configure error which goes away without that argument: 
```log
CMake Error at /bld4/opt/cmake/3.23.2/share/cmake-3.23/Modules/ExternalProject.cmake:2670 (message):
  At least one entry of URL is a path (invalid in a list)
Call Stack (most recent call first):
  /bld4/opt/cmake/3.23.2/share/cmake-3.23/Modules/ExternalProject.cmake:3640 (_ep_add_download_command)
  CMakeLists.txt:15 (ExternalProject_Add)


-- Configuring incomplete, errors occurred!
See also "/bld4/home/esseivaj/devel/celeritas/build/_deps/perfetto-subbuild/CMakeFiles/CMakeOutput.log".

CMake Error at /bld4/opt/cmake/3.23.2/share/cmake-3.23/Modules/FetchContent.cmake:1076 (message):
  CMake step for perfetto failed: 1
Call Stack (most recent call first):
  /bld4/opt/cmake/3.23.2/share/cmake-3.23/Modules/FetchContent.cmake:1217:EVAL:2 (__FetchContent_directPopulate)
  /bld4/opt/cmake/3.23.2/share/cmake-3.23/Modules/FetchContent.cmake:1217 (cmake_language)
  /bld4/opt/cmake/3.23.2/share/cmake-3.23/Modules/FetchContent.cmake:1260 (FetchContent_Populate)
  CMakeLists.txt:353 (FetchContent_MakeAvailable)
```

So we probably want to set the policy a policy you suggested:
```cmake
if(${CMAKE_VERSION} VERSION_GREATER_EQUAL 3.24)
  cmake_policy(SET CMP0135 OLD)
endif()
```

Reply:
Thanks!

Reply:
Actually I'm going to take the opportunity to update the policy version range in the top-level header and require CMake 3.18.

Reply:
we still need to set the policy to the OLD behavior if we want ```DOWNLOAD_EXTRACT_TIMESTAMP TRUE``` for all versions of CMake

Reply:
I guess I don't really care if the default behavior changes, I was mostly trying to silence the warning